app/mailers/user_notifications.rb

```diff
@@ -433,6 +443,14 @@ def send_notification_email(opts)
       show_category_in_subject = nil
     end
 
+    # tag names
+    tags = Topic.find_by(id: post.topic_id)&.tags
+    if opts[:show_tags_in_subject] && post.topic_id && tags
```

Comment:
I would first check tags are allowed in subject and only then do the SQL query.

Reply:
Also, there should probably be a limit of the number of tags retrieved.

Reply:
What would be the best limit? 3 tags? Should there be any specific order as well then?

Reply:
3 seems fine. I would keep the same order as in the post.